apps/condo/domains/common/utils/serverSchema/changeTrackable.js

```diff
@@ -0,0 +1,355 @@
+                })
+                if (_.difference(existing.ids, updated.ids).length > 0) {
+                    data[`${ key }IdsFrom`] = existing.ids
+                    data[`${ key }IdsTo`] = updated.ids
+                    data[`${ key }DisplayNamesFrom`] = existing.displayNames
+                    data[`${ key }DisplayNamesTo`] = updated.displayNames
+                }
+
+            }
+        }
+    }))
+    return data
+}
+
+const isScalar = (field) => (
+    field.type !== Relationship
+)
+
+const isRelationSingle = (field) => (
+    field.type === Relationship && !field.many
+)
+
+const isRelationMany = (field) => (
+    field.type === Relationship && field.many
+)
+
+const mapScalars = (acc, value, key) => {
+    acc[`${key}From`] = mapScalar(value)
+    acc[`${key}To`] = mapScalar(value)
+}
+
+const mapScalar = (field) => (
+    _.pick(field, ['schemaDoc', 'type'])
+)
+
+/**
+ * Produces "Change storage set" of fields (see Terms) for a single relationship field
+ * Used in lodash `transform` function
+ * @param {Object} acc - final set all fields, composed using lodash `transform` function
+ * @param {Object} value - Keystone field declaration
+ * @param {String} key - key of a field being iterated
+ */
+const mapRelationSingle = (acc, value, key) => {
+    acc[`${key}IdFrom`] = {
+        schemaDoc: `Old id of related entity. ${value.schemaDoc}`,
+        type: Uuid,
+    }
+    acc[`${key}IdTo`] = {
+        schemaDoc: `New id of related entity. ${value.schemaDoc}`,
+        type: Uuid,
+    }
+    acc[`${key}DisplayNameFrom`] = {
+        schemaDoc: `Old display name of related entity. ${value.schemaDoc}`,
+        type: Text,
+    }
+    acc[`${key}DisplayNameTo`] = {
+        schemaDoc: `New display name of related entity. ${value.schemaDoc}`,
+        type: Text,
+    }
+}
+
+/**
+ * Produces "Change storage set" of fields (see Terms) for a "many" relationship field
+ * Used in lodash `transform` function
+ * @param {Object} acc - final set all fields, composed using lodash `transform` function
+ * @param {Object} value - Keystone field declaration
+ * @param {String} key - key of a field being iterated
+ */
+const mapRelationMany = (acc, value, key) => {
+    acc[`${key}IdsFrom`] = {
+        schemaDoc: `Old list of ids of related entities. ${value.schemaDoc}`,
+        type: Json,
+        defaultValue: [],
+    }
+    acc[`${key}IdsTo`] = {
+        schemaDoc: `New list of ids of related entities. ${value.schemaDoc}`,
+        type: Json,
+        defaultValue: [],
+    }
+    acc[`${key}DisplayNamesFrom`] = {
+        schemaDoc: `Old version of display names of related entities. ${value.schemaDoc}`,
+        type: Json,
+        defaultValue: [],
+    }
+    acc[`${key}DisplayNamesTo`] = {
+        schemaDoc: `New version of display names of related entities. ${value.schemaDoc}`,
+        type: Json,
+        defaultValue: [],
+    }
+}
+
+module.exports = {
+    trackableFieldsFrom,
```

Comment:
best practice to actually tell what the function is doing by giving the function name a verb as prefix

Reply:
Fixed in `673b318b`